### ISSUE TITLE
Git reset during the maven release

### ISSUE THREAD

**Issue by sergioasantiago** [qcastel/github-actions-maven-release#31]
I'm trying to run the action in my project, but I would like to use a different commit than the one that triggered the action.
I have a monorepo project, and a matrix strategy to create multiple releases, therefore, once I try to create a release after a previous one has been released, it throws the exception:

```
Error: 5,300 [ERROR]  ! [rejected]        main -> main (non-fast-forward)
Error: 5,300 [ERROR] error: failed to push some refs to 'https://github.com/organization/project.git'
Error: 5,301 [ERROR] hint: Updates were rejected because the tip of your current branch is behind
Error: 5,301 [ERROR] hint: its remote counterpart. Integrate the remote changes (e.g.
Error: 5,301 [ERROR] hint: 'git pull ...') before pushing again.
Error: 5,301 [ERROR] hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Checking the logs, I realized that the action is forcing a reset as per logs:

```
Last *** commit: 6d7eec8fcc7c1d4a8a3e37a1846f2b6f2dd2a52e
Current commit: 764bcf1b0ce52dea3f61331a695cd571aa17d736
INFO - M2 folder '/root/.m2' not empty. We therefore will beneficy from the CI cache
Current branch: main
Release branch name: main
We are on the release branch
Using SSH folder /root/.ssh
Using M2 repository folder /root/.m2
/root/.m2 is empty, this means we didn't hit a potential M2 cache :(
Git checkout branch main
Already on 'main'
Your branch is up to date with 'origin/main'.
Git reset hard to 764bcf1b0ce52dea3f61331a695cd571aa17d736
```

Do you have any idea why?

**Comment by qcastel** [comment]
hello @sergioasantiago ,

Is it possible to see the repo that you are working on? it would help me visualise what is your scenario exactly.

In the meantime, I can give you some indication on how the maven: You can find the script that is used for doing the maven release. As you found out, there is indeed a git reset to assure we are on top of master before proceeding to the rest of the flow.
https://github.com/qcastel/docker-maven-release/blob/master/release.sh#L54


My first intuition would be perhaps to use the filtering system of github actions to select exactly the kind of commits that should trigger the release. A bit like I did here for the sample repo: https://github.com/qcastel/github-actions-maven-release-sample/blob/master/.github/workflows/release.yaml#L11

**Comment by sergioasantiago** [comment]
Hey! Thank you for your reply.

The repo is: https://github.com/freenowtech/sauron-core

> As you found out, there is indeed a git reset to assure we are on top of master before proceeding to the rest of the flow.
https://github.com/qcastel/docker-maven-release/blob/master/release.sh#L54

Do you think it is worth it to make this a feature flag?

I can't think in a way where the filters would help in my scenario. If you check in the repo above, [here](https://github.com/freenowtech/sauron-core/blob/main/.github/workflows/release.yml#L28) I'm using the strategy matrix, which will create concurrent jobs for each entry, and therefore one output per entry, which leads to the commits between the jobs. 

[Here](https://github.com/freenowtech/sauron-core/blob/main/.github/workflows/release.yml#L52) I do a git pull to make sure I pick up the latest commits, which I agree can be dangerous, but I'm prepared to live with that.

**Comment by qcastel** [comment]
@sergioasantiago a feature flag is reasonable, we got to make it flavour for everyone dont we?
Anyway you seems to make sure of the state of the repo before running the action, should be fine then.

https://github.com/qcastel/github-actions-maven-release/blob/master/action.yml#L42

Here is the flag that you can use, let me know if that solve your problem.

**Comment by sergioasantiago** [comment]
@qcastel  thanks for your implementation!
I was checking the release.sh and since it is using `if [[ -z "${SKIP_GIT_SANITY_CHECK}" ]]; then` if I set the value of `git-skip-sanity-check` to `true`, then I think this if will not do what we expect. I tried it and it still executing the reset.

Am I'm doing something wrong?

**Comment by sergioasantiago** [comment]
Hey, no problem at all. Rest is always important. I hope you had great time.
With my workaround it has been work last days.
I will try the new version in the next days and let you know.
Thanks again!

**Comment by sergioasantiago** [comment]
Hey I was about to try the new version, but I could not find a new version, only the one that I'm using 😢

**Comment by qcastel** [comment]
@sergioasantiago I have rested enough, you deserve now that I tackle this. Apologies again for the delay.

I released https://github.com/qcastel/github-actions-maven-release/releases/tag/v1.12.17 
This time, I also tried on the sample repo and can confirm it will work.
Just do that: https://github.com/gh-a-sample/github-actions-maven-release-sample/commit/0fa8b3ecc441e920423b37c038c60c2afdae1300
and you will be fine.

I keep this one open until you confirm it's all good for you.
